spec/javascripts/controllers/middleware/middleware_topology_controller_spec.js

```diff
@@ -65,28 +73,42 @@ describe('middlewareTopologyController', function () {
     });
   });
 
+
   describe('kinds contains all expected mw kinds', function () {
     it('in all main objects', function () {
-      expect(Object.keys(scope.kinds).length).toBe(3);
+      expect(Object.keys(scope.kinds).length).toBe(4);
       expect(scope.kinds["MiddlewareManager"]).toBeDefined();
       expect(scope.kinds["MiddlewareServer"]).toBeDefined();
       expect(scope.kinds["MiddlewareDeployment"]).toBeDefined();
+      expect(scope.kinds["MiddlewareDatasource"]).toBeDefined();
     });
   });
 
+
   describe('the mw topology gets correct icons', function () {
     it('in graph elements', function () {
       expect($controller.class_name(mw_manager)).toContain("vendor-hawkular");
       expect($controller.class_name(mw_server)).toContain("vendor-hawkular");
       expect($controller.class_name(mw_deployment)).toContain("middleware_deployment_war");
+      expect($controller.class_name(mw_datasource)).toBeUndefined()
```

Comment:
I am following the link to the code diff, but it looks like it's bringing me to an unrelated place, or I don't understand the relation to the code in the test above.